rai/client.go

```diff
@@ -1836,7 +1838,75 @@ func (c *Client) GetSnowflakeDataStreamStatus(
 	integration, dbLink, objectName string,
 ) (*SnowflakeDataStreamStatus, error) {
 	var result SnowflakeDataStreamStatus
-	path := makePath(PathIntegrations, integration, "database-links", dbLink, "data-streams", objectName, "status")
+	path := makePath(PathIntegrationsAlpha, integration, "database-links", dbLink, "data-streams", objectName, "status")
+	if err := c.Get(path, nil, nil, &result); err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+// beta
+
+type DataStreamOpts struct {
+	RaiDatabase string
+	Relation    string
+	ObjectName  string
+}
+
+// Register snowflake datastream created by native app to replicate data from a Snowflake table/view to a RAI relation.
+func (c *Client) RegisterSnowflakeDataStream(
```

Comment:
are these intended for a customer or RAI use? does the caller need a rai token from the linked account?

Reply:
These are intended for RAI's use; the proxy will call into the integration service endpoint using these.

Reply:
whats is preventing client users form calling these endpoints? seems like a potential risk

Reply:
They can't call these endpoints without an RAI token. To generate the token, they need to know the account associated with the integration and have access to that account. Please correct me if I am missing a point here.